File: LeetCode/0020-valid-parentheses/0020-valid-parentheses.c
bool isValid(char* s) {
    char stack[100000];
    int top = -1;

    for(int i = 0; s[i] != '\0'; ++i){
        char c = s[i];

        if(c == '(' || c == '{' || c == '['){
            ++top;
            stack[top] = c;
        }
        
        else{
            if(top == -1){
                return false;
            }

            if(c == ')' && stack[top] != '(' || 
               c == '}' && stack[top] != '{' ||
               c == ']' && stack[top] != '['){
                return false;
            }

            top--;
        }
    }

    if(top == -1){
        return true;
    }

    else{
        return false;
    }
}

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []

        for char in s:
            if char == '(':
                stack.append('(')

            elif char == '{':
                stack.append('{')   

            elif char == '[':
                stack.append('[')    


            else:
                if len(stack):
                    return False
                
                top = stack.pop()

                if char == ')' and top != '(':
                    return False
                elif char == '}' and top != '{':
                    return False
                elif char == ']' and top != '[':
                    return False

        if len(stack) == 0:
            return True 
        else:
            return False